### PR TITLE
Use signatures in OpenQA::Schema::Result::Jobs

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -360,16 +339,12 @@ sub reschedule_state {
     log_debug('Job ' . $self->id . " reset to state $state");
 
     # free the worker
-    if (my $worker = $self->worker) {
-        $self->worker->update({job_id => undef});
-    }
+    if (my $worker = $self->worker) { $self->worker->update({job_id => undef}) }
```
Comment (perlpunk) [inline lib/OpenQA/Schema/Result/Jobs.pm:342]: This looks like it was already intended to do this in the original code:
```suggestion
    if (my $worker = $self->worker) { $worker->update({job_id => undef}) }
```

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 2]
```diff
@@ -1512,14 +1424,9 @@ sub create_asset {
...
 }
 
-sub has_failed_modules {
-    my ($self) = @_;
-    return $self->modules->count({result => 'failed'}, {rows => 1});
-}
-
-sub failed_modules {
-    my ($self) = @_;
+sub has_failed_modules ($self) { $self->modules->count({result => 'failed'}, {rows => 1}) }
```
Comment (perlpunk) [inline lib/OpenQA/Schema/Result/Jobs.pm:1427]: `rows => 1` is s no-op for `count()`

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 3]
```diff
@@ -1887,18 +1766,14 @@ sub store_column {
     return $self->SUPER::store_column($columnname, $value);
 }
 
-sub enqueue_finalize_job_results {
-    my ($self, $carried_over) = @_;
-
+sub enqueue_finalize_job_results ($self, $carried_over) {
```
Comment (perlpunk) [inline lib/OpenQA/Schema/Result/Jobs.pm:1769]:
```suggestion
sub enqueue_finalize_job_results ($self, $carried_over = undef) {
```

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 1 above)
Comment (okurz) [reply lib/OpenQA/Schema/Result/Jobs.pm]: done

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 2 above)
Comment (okurz) [reply lib/OpenQA/Schema/Result/Jobs.pm]: done

File: lib/OpenQA/Schema/Result/Jobs.pm (hunk 3 above)
Comment (okurz) [reply lib/OpenQA/Schema/Result/Jobs.pm]: done

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 4]
```diff
@@ -857,9 +813,7 @@ sub cluster_jobs {
 }
 
 # internal (recursive) function used by cluster_jobs to invoke itself for all children
-sub _cluster_children {
-    my ($self, $jobs, $skip_parents, $no_directly_chained_parent) = @_;
-
+sub _cluster_children ($self, $jobs, $skip_parents, $no_directly_chained_parent) {
```
Comment (perlpunk) [inline lib/OpenQA/Schema/Result/Jobs.pm:816]:
```suggestion
sub _cluster_children ($self, $jobs, $skip_parents = undef, $no_directly_chained_parent = undef) {
```

Comment (okurz) [reply lib/OpenQA/Schema/Result/Jobs.pm:816]: done

**Review by perlpunk — APPROVED** [review]

**Review by Martchus — APPROVED** [review]

**Pull request merged**